text/src/autogluon/text/text_prediction/mx/models.py

```diff
@@ -0,0 +1,1505 @@
+    else:
+        raise ValueError('Unsupported lr_scheduler="{}"'
+                         .format(cfg.lr_scheduler))
+    optimizer_params = {'learning_rate': cfg.lr,
+                        'wd': cfg.wd,
+                        'lr_scheduler': lr_scheduler}
+    optimizer = cfg.optimizer
+    additional_params = {key: value for key, value in cfg.optimizer_params}
+    optimizer_params.update(additional_params)
+    return optimizer, optimizer_params, max_update
+
+
+@use_np
+def apply_layerwise_decay(model, layerwise_decay, backbone_name, not_included=None):
+    """Apply the layer-wise gradient decay
+
+    .. math::
+        lr = lr * layerwise_decay^(max_depth - layer_depth)
+
+    Parameters:
+    ----------
+    model
+        The backbone model
+    layerwise_decay: int
+        layer-wise decay power
+    not_included: list of str
+        A list or parameter names that not included in the layer-wise decay
+    """
+    if not_included is None:
+        not_included = []
+    # consider the task specific fine-tuning layer as the last layer, following with pooler
+    # In addition, the embedding parameters have the smaller learning rate based on this setting.
+    if 'albert' in backbone_name:
+        # Skip if it is the ALBERT model.
+        return
+    if 'electra' in backbone_name:
+        # For ELECTRA, it's called all_encoder_layers
+        all_layers = model.encoder.all_encoder_layers
+    else:
+        # For other models, it's called all_layers
+        all_layers = model.encoder.all_layers
+    max_depth = len(all_layers) + 2
+    for key, value in model.collect_params().items():
+        if 'scores' in key:
+            value.lr_mult = layerwise_decay ** 0
+        if 'pool' in key:
+            value.lr_mult = layerwise_decay ** 1
+        if 'embed' in key:
+            value.lr_mult = layerwise_decay ** max_depth
+
+    for (layer_depth, layer) in enumerate(all_layers):
+        layer_params = layer.collect_params()
+        for key, value in layer_params.items():
+            for pn in not_included:
+                if pn in key:
+                    continue
+            value.lr_mult = layerwise_decay ** (max_depth - (layer_depth + 1))
+
+
+@use_np
+def freeze_layers(model, backbone_name, num_trainable_layers):
+    if 'albert' in backbone_name:
+        # Skip if it is the ALBERT model.
+        return
+    if 'electra' in backbone_name:
+        # For ELECTRA, it's called all_encoder_layers
+        all_layers = model.encoder.all_encoder_layers
+    else:
+        # For other models, it's called all_layers
+        all_layers = model.encoder.all_layers
+    if num_trainable_layers < 0:
+        return
+    assert num_trainable_layers <= len(all_layers)
+    for i in range(len(all_layers) - num_trainable_layers):
+        for p in all_layers[i].collect_params().values():
+            p.grad_req = 'null'
+    return
+
+
+def base_optimization_config():
+    """The basic optimization phase"""
+    cfg = CfgNode()
```

Comment:
Is the optimizer momentum customizable? [Paper](https://arxiv.org/pdf/1803.09820.pdf)
> Momentum:  Short runs with momentum values of 0.99, 0.97, 0.95, and 0.9 will quicklyshow the best value for momentum. If using the 1cycle learning rate schedule, it is better touse a cyclical momentum (CM) that starts at this maximum momentum value and decreaseswith increasing learning rate to a value of 0.8 or 0.85 (performance is almost independentof the minimum momentum value).  Using cyclical momentum along with the LR rangetest stabilizes the convergence when using large learning rate values more than a constantmomentum does.

Reply:
Yes, just need to change these values.

Reply:
We can specify the value in presets like in http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-936/77/tutorials/text_prediction/customization.html#change-hyperparameter

Reply:
Basically, all data in the yml config are configurable.